### DISCUSSION TITLE
Hubot and mattermost adapter

### DISCUSSION THREAD

**Discussion by EbsrVlee** [hubotio/hubot#1593] [Q&A]
Hello All,

Not sure if this is the correct place to post this.
I am trying to setup hubot with the mattermost adapter but am having some issues. I run the “yo hubot” command and it asks me for the adapter, I put mattermost but then ran into the warning "hubot-help@0.2.2 requires a peer of coffee-script@^1.12.6 but none was installed. I then ran npm install -g coffee-script and coffee-script 1.12.7 was installed.

I then go back to run hubot again and get the warning “UNMET PEER DEPENDENCY coffee-script@^1.12.6” and I see the adapter is still campfire not mattermost.

I am not sure how to proceed. Does anyone have any advice?

Thank you.

- Victoria

**Comment by bafplus** [comment]
First run yo hubot and choose "shell" as adapter.
This wil setup the basic bot.
Then you need to install and configure a mattermost adapter, like this one: 
npm install hubot-mattermost-adapter --save

After install you can then run that hubot with the mattermost adapter:
./bin/hubot --adapter mattermost

**Comment by EbsrVlee** [comment]
Thank you @bafplus , for your suggestion.
I followed those steps and afterwards when I ask hubot what it's adapter is, it says "shell". The mattermost adapter doesn't seem to be saving.

**Comment by bafplus** [comment]
Then you probably started the hubot with the shell adapter instead of stating the mattermost adapter.
I just noticed i added th wrong adapter name so heres a new one:

./bin/hubot --adapter mattermost-adapter

**Comment by EbsrVlee** [comment]
Interesting. When I run ./bin/hubot --adapter mattermost-adapter I get the message "EMERGENCY MATTERMOST_HOST is required" 

I then ask hubot for it's adapter and it's blank instead of saying "shell" like before.

**Comment by bafplus** [comment]
That means the command is correct now but you havent supplied the (correct) environment variables.

https://www.npmjs.com/package/hubot-mattermost-adapter

I cant help you with that, maybe somebody else here?

**Comment by EbsrVlee** [comment]
@bafplus Thank you so much for your help so far!

**Comment by EbsrVlee** [comment]
Hello, does anyone else have any more advice? I tried running ./bin/hubot --adapter mattermost-adapter and ended up with error login call failed. If I run ./bin/hubot I get the name of my hubot. Then when I ask for adapter it is still stuck as shell.

![mattermostadapter](https://user-images.githubusercontent.com/14913403/33450933-5ca931d8-d5db-11e7-8adc-a6c0111158e6.jpg)
